test/e2e/framework/azure/vm_manager.go

```diff
@@ -53,7 +53,15 @@ func (m *vmManager) AssignUserAssignedIdentity(vmName, identityToAssign string)
 		}
 	}
 
-	vm.Identity.UserAssignedIdentities[fmt.Sprintf(ResourceIDTemplate, m.config.SubscriptionID, m.config.IdentityResourceGroup, identityToAssign)] = &compute.VirtualMachineIdentityUserAssignedIdentitiesValue{}
+	identityAssignResourceID := fmt.Sprintf(ResourceIDTemplate, m.config.SubscriptionID, m.config.IdentityResourceGroup, identityToAssign)
+	for identity := range vm.Identity.UserAssignedIdentities {
+		// identity already exists and doesn't need to be re-assigned
+		if strings.EqualFold(identity, identityAssignResourceID) {
+			return nil
+		}
+	}
+
+	vm.Identity.UserAssignedIdentities[identityAssignResourceID] = &compute.VirtualMachineIdentityUserAssignedIdentitiesValue{}
```

Comment:
Got the following error https://dev.azure.com/azure/aad-pod-identity/_build/results?buildId=12327&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=c0307c8e-cb29-5919-a192-498b9fd8bfd8&l=217 in #687. Do you know if it's possible to have `vm.Identity`not equal to nil but `vm.Identity.UserAssignedIdentities` is nil?

Reply:
It is possible if the SDK is setting to nil. In our actual code path, we do check if its nil and initialize if that's the case. Let me update this PR to do that.